viewing_party/party.py

```diff
@@ -1,23 +1,135 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    if not title or not genre or not rating:
+        return None
+    new_movie = {}
+    new_movie["title"] = title
+    new_movie["genre"] = genre
+    new_movie["rating"] = rating
+    return new_movie
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data, title):
+    watched_movie = user_data["watched"] # [{}, {}]
+    watchlist = user_data["watchlist"] 
+    for movie in watchlist:
+        if movie["title"] == title:
+            watched_movie.append(movie)
+            watchlist.remove(movie)
+    return user_data
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
+def get_watched_avg_rating(user_data):
+    watched_movies = user_data["watched"]

```

Comment:
👍 Nice temporary variable to refer to the nested watched list more easily. While we could have written out `user_data["watched"]` everywhere, this would be repetitive and error prone.